rts/Sim/Weapons/WeaponDef.cpp

```diff
@@ -390,7 +386,7 @@ WeaponDef::WeaponDef(const LuaTable& wdTable, const std::string& name_, int id_)
 			const int type = damageArrayHandler.GetTypeFromName(pair.first);
 			if (type == 0)
 				continue;
-			damages.Set(type, std::max(0.0001f, pair.second));
+			damages.Set(type, pair.second);
```

Comment:
Consider switching to https://en.cppreference.com/w/cpp/language/structured_binding for loop arguments here.